shared/binlog.c

```diff
@@ -589,12 +573,19 @@ binlog * binlog_get_saved(binlog * node_binlog) {
 
 	/* free'd either here (binlog_destroy) or by the caller (node_binlog_read_saved) */
 	bl = malloc(sizeof(struct binlog));
+	if (bl == NULL) {
+		fclose(file);
+		return NULL;
+	}
+	memset(bl, 0, sizeof(struct binlog));
 
 	elements_read = fread(bl, sizeof(struct binlog), 1, file);
 	fclose(file);
 
 	/* Make sure we sucessfully read one binlog struct */
 	if (elements_read != 1) {
+		unlink(node_binlog->file_save_path);
+		unlink(node_binlog->file_metadata_path);
 		binlog_destroy(bl,BINLOG_UNLINK);
```

Comment:
Hmmm. After some checking,  binlog_destroy does multiple processing on bl. However, the bl read was invalid. Does it make sense to just remove the destroy here and just free bl?

Reply:
that's actually a good point and much easier to read in this case so I agree. Let me just make sure the return value of fread for different scenarios.

Reply:
Replaced call to binlog_destroy with free.